src/Course2/week1/ConnectedComponents.java

```diff
@@ -0,0 +1,69 @@
+package Course2.week1;
+
+import java.util.ArrayList;
+
+public class ConnectedComponents {
+     static void DFSRec(ArrayList<ArrayList<Integer>> adj,int s,boolean[] visited){
+
+            visited[s]=true;
+//            System.out.print(s+" ");
+
+            for(int u:adj.get(s)){
+                if(visited[u] == false){
+                    DFSRec(adj,u,visited);
+                }
+            }
+        }
+
+        static int DFS(ArrayList<ArrayList<Integer>> adj,int V,int s){
+
+            int countComponents = 0;
+            boolean[] visited = new boolean[V];
+
+            for(int i=0;i<V;i++)
+                visited[i]=false;
+
+            for(int i=0;i<V;i++){
+                if(visited[i]==false){
+                    DFSRec(adj,i,visited);
+                    countComponents++;
+                }
+            }
+            return countComponents;
+        }
+
+        static void printGraph(ArrayList<ArrayList<Integer>> adj){
+            for(int i=0;i< adj.size();i++){
+                for(int j=0;j<adj.get(i).size();j++){
+                    System.out.print(adj.get(i).get(j)+" ");
+                }
+                System.out.println();
+            }
+        }
+
+        static void addedge(ArrayList<ArrayList<Integer>> adj,int u ,int v){
+            adj.get(u).add(v);
+            adj.get(v).add(u);
+        }
+
+        public static void main(String[] args){
+            int V=7;
+
```

Comment:
excessive use of nested for loops and in initialization 
wrong method naming convention
formatting issues